# Step 1: State the Hypotheses
Null Hypothesis (H0): There is no association between device type and customer satisfaction.

Alternative Hypothesis (H1): There is an association between device type and customer satisfaction.

In [ ]:
import scipy.stats as stats
data = [
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
]

# Step 2: Compute the Chi-Square Statistic


In [ ]:
chi2_stat, p_value, _, _ = stats.chi2_contingency(data)

# Step 3: Determine the Critical Value


In [ ]:
alpha = 0.05
degrees_of_freedom = (len(data) - 1) * (len(data[0]) - 1)
critical_value = stats.chi2.ppf(1 - alpha, degrees_of_freedom)

# Step 4: Make a Decision


In [ ]:
if chi2_stat > critical_value:
    decision = "Reject the null hypothesis"
else:
    decision = "Fail to reject the null hypothesis"

print("Chi-Square Statistic:", chi2_stat)
print("P-Value:", p_value)
print("Critical Value:", critical_value)
print("Decision:", decision)

if decision == "Reject the null hypothesis":
    print("There is a significant association between device type and customer satisfaction.")
else:
    print("There is no significant association between device type and customer satisfaction.")

Chi-Square Statistic: 5.638227513227513
P-Value: 0.22784371130697179
Critical Value: 9.487729036781154
Decision: Fail to reject the null hypothesis
There is no significant association between device type and customer satisfaction.


Chi-Square Test for Independence
Observed Data:
 [[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]

 Expected Frequencies:
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]

Chi-Square Statistic: 5.638227513227513

Degrees of Freedom: 4

Critical Value: 9.487729036781154

P-value: 0.22784371130697179

Reject the null hypothesis: False


Conclusion: There is no significant association between the type of device and customer satisfaction level.